spec/models/author_spec.rb

```diff
@@ -0,0 +1,88 @@
+require 'rails_helper'
+
+RSpec.describe Author, type: :model do
```

Comment:
you don't need to write so many code for model testing. Just do smth like this:
```ruby
RSpec.describe Author, type: :model do
  context 'validations' do
    it { is_expected.to validate_presence_of(:first_name) }
    it { is_expected.to validate_presence_of(:last_name) }
  end

  context 'associations' do
    it { is_expected.to have_many(:authors_books).dependent(:destroy) }
    it { is_expected.to have_many(:books).through(:authors_books) }
  end
end
```
The same about all your models tests